api/src/main/java/com/cloud/resource/ResourceState.java

```diff
@@ -22,17 +22,26 @@
 import com.cloud.utils.fsm.StateMachine;
 
 public enum ResourceState {
-    Creating, Enabled, Disabled, PrepareForMaintenance, ErrorInMaintenance, Maintenance, Error;
+    Creating,
+    Enabled,
+    Disabled,
+    PrepareForMaintenance,
+    ErrorInMaintenance,
+    Maintenance,
+    Error,
+    PrepareForMaintenanceErrorsPresent;
```

Comment:
Can you rename `PrepareForMaintenanceErrorsPresent` to `ErrorInPrepareForMaintenance` to be uniform with other names such as `ErrorInMaintenance`.

Reply:
PrepareForMaintenanceErrorsPresent is a transitional state that can go back to PrepareForMaintenance if Admin fixes errors.

This ultimately ends in ErrorInMaintenance or Maintenance state which are final states to be reached. Let me update the PR description @rhtyd

Reply:
(minor nit)

I understand, if the intent of the new state to say errors happened while doing `...` then I'm simply asking the name of the new state to be uniform with other state names. For example, `Maintenance` has `ErrorInMaintenance`; so `PrepareForMaintenance` can have a similar `ErrorInPrepareForMaintenance` state.

Reply:
Okay. I get your point. Fixing in the next commit.